### PR TITLE
Fix NSURLConnection is deprecated - Use NSURLSession

### PR DESCRIPTION
**Pull request by Coeur**
Problem: NSURLConnection is deprecated since macOS 10.11
Solution: Use NSURLSession

### CODE REVIEW

File: macosx/PortChecker.mm [hunk 1]
```diff
@@ -112,30 +66,48 @@ - (void)startProbe:(NSTimer*)timer
...
-    }
+    _fTask = [_fSession dataTaskWithRequest:portProbeRequest
+                          completionHandler:^(NSData* _Nullable data, NSURLResponse* _Nullable response, NSError* _Nullable error) {
+                              if (error)
+                              {
+                                  NSLog(@"Unable to get port status: connection failed (%@)", error.localizedDescription);
+                                  [self callBackWithStatus:PORT_STATUS_ERROR];
+                                  return;
+                              }
+                              NSString* probeString = [[NSString alloc] initWithData:data ?: NSData.data encoding:NSUTF8StringEncoding];
```
Comment (nevack) [inline macosx/PortChecker.mm:82]: Obj-C has elvis operator?

Comment (Coeur) [reply macosx/PortChecker.mm:82]: Yes, it's GNU C, Objective-C, Objective-C++, available with both GCC and clang.

**Review by nevack — APPROVED** [review]
LGTM, thanks @Coeur

**Comment by ckerr** [comment]
Notes: Fixed macOS API deprecation warnings.

**Pull request merged**